process/proofProcessor.go

```diff
@@ -0,0 +1,140 @@
+package process
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+)
+
+type ProofProcessor struct {
+	proc            Processor
+	pubKeyConverter core.PubkeyConverter
+}
+
+func NewProofProcessor(proc Processor, pubKeyConverter core.PubkeyConverter) (*ProofProcessor, error) {
+	if check.IfNil(proc) {
+		return nil, ErrNilCoreProcessor
+	}
+	if check.IfNil(pubKeyConverter) {
+		return nil, ErrNilPubKeyConverter
+	}
+
+	return &ProofProcessor{
+		proc:            proc,
+		pubKeyConverter: pubKeyConverter,
+	}, nil
+}
+
+// GetProof sends the request to the right observer and then replies with the returned answer
+func (pp *ProofProcessor) GetProof(rootHash []byte, address []byte) ([][]byte, error) {
```

Comment:
if you don't need to change in any way the response from the node, then you could use GenericApiResponse and this way you make sure the response is consistent with the one from the node

Reply:
Done.